hydra/main.py

```diff
@@ -42,15 +71,23 @@ def decorated_main(cfg_passthrough: Optional[DictConfig] = None) -> Any:
             if cfg_passthrough is not None:
                 return task_function(cfg_passthrough)
             else:
-                args = get_args_parser()
-                # no return value from run_hydra() as it may sometime actually run the task_function
-                # multiple times (--multirun)
-                _run_hydra(
-                    args_parser=args,
-                    task_function=task_function,
-                    config_path=config_path,
-                    config_name=config_name,
-                )
+                args_parser = get_args_parser()
+                args = args_parser.parse_args()
+                if args.experimental_rerun is not None:
+                    cfg = _get_rerun_conf(args.experimental_rerun, args.overrides)
+                    ret = task_function(cfg)
+                    _flush_loggers()
+                    return ret
```

Comment:
Why return the value here? The `else` branch below does not return anything.

Reply:
returnting here to be consistent with the `cfg_passthrough` case above, since this feels more like a special case of `cfg_passthrough`...

Reply:
Hmm, ok.

Reply:
What if we support `--experimental-rerun` in combination with the `--multirun` flag in the future (so that users could pass a sweep override)? It's not clear what the return value would be. This is the reason why the `else` branch below does not return a value (see the comment on that branch).

Reply:
Good point. `multirun` is going to be way more complex than what is purposed here. And as I mentioned [here](https://github.com/facebookresearch/hydra/issues/1805#issuecomment-1059600785) I think it makes sense to only support SINGLE run for now and see how it goes. I should have make it more clear on what this issue is trying to address, I've updated the PR description. 

I will remove the return statement since I agree it sees redundant.